src/ninja.cc

```diff
@@ -945,6 +948,32 @@ void printCompdb(const char* const directory, const Edge* const edge,
   printf("\"\n  }");
 }
 
+struct DependentEdgeCollector {
```

Comment:
Please move this class definition to `src/graph.h` and provide proper documentation + unit-tests for it. See `InputsCollector` there for inspiration.

I also recommend to make `depend_edges` a member of the class, and add a method like `TakeResult()` to move the final result out of the instance one all root targets have been visited.

Reply:
> Please move this class definition to `src/graph.h`

is it ok if i move it to a new header file instead?  It's very `compdb`-specific, and only used by `ninja.cc` and `graph_test.cc`, so it would be nice if changing it didn't cause a lot of unrelated source files to be recompiled